### DISCUSSION TITLE
Temperature AMD 7800X3D on Linux

### DISCUSSION THREAD

**Discussion by chrizdeluxe** [mathoudebine/turing-smart-screen-python#393] [General]
Hi,
I'm just starting to tinkering around with themes and i recognized that the CPU temperature from my CPU is not showing correctly.

I think its not a real issue so i post it here first.

Its showing the Tctl instead of the Tccd1.
Tctl has as far as i know an offset and its not the real temperature of the CPU.
On older AMD CPUs there is a Tdie on the first position.

This is the output of lm_sensors, i think psutil get this in the same order.
> k10temp-pci-00c3
Adapter: PCI adapter
Tctl:         +45.5°C  
Tccd1:        +41.6°C  

After some research in the code i found where the CPU temperature is read and its the first position.

https://github.com/mathoudebine/turing-smart-screen-python/blob/d44505f03ab7f35cbb9f7eb9bf3bed86d182d458/library/sensors/sensors_python.py#L85-L94

My temporarily solution is to edit line 94 to
`cpu_temp = sensors_temps['k10temp'][1].current`

Does somebody know if there is no Tdie for the newer AMD CPUs or am i missing something to see the Tdie?
I'm running Kernel 6.6.1-arch1-1

Beside that 'issue' it runs pretty good, thanks for your great work.

**Comment by marcosins** [comment]
My 7800x3D too doesn't report Tdie sensor

```bash
$ uname -mr && sensors && lscpu -J | jq '.["lscpu"][7].data'
6.8.0-49-generic x86_64
(...)
k10temp-pci-00c3
Adapter: PCI adapter
Tctl:         +47.1°C  
Tccd1:        +42.6°C  

"AMD Ryzen 7 7800X3D 8-Core Processor"
```

Edit: found this extract from the 5.15 release[^1][^2]
> Other notable changes:
> - In k10temp, do not display Tdie on Zen CPUs if there is no difference between Tdie and Tctl
[^1]: https://lkml.iu.edu/hypermail/linux/kernel/2108.3/06462.html#:~:text=difference%20between%20Tdie%20and%20Tctl
[^2]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=e7c1bbcf0c315c56cd970642214aa1df3d8cf61d

**Comment by NeyMendesJr** [comment]
I'm new to Python and I can't seem to get the temperature reading to work for my AMD 9600x CPU. Everything else in my code is running fine.
